src/main/kotlin/ar/edu/unahur/obj2/servidorWeb/Analizador.kt

```diff
@@ -0,0 +1,71 @@
+package ar.edu.unahur.obj2.servidorWeb
+
+import java.time.LocalDateTime
+import java.util.*
+
+abstract class Analizador() {
+    val respuestas = mutableListOf<Respuesta>()
+    val modulos = mutableListOf<Modulo?>()
+
+    open fun procesar(respuestaProcesada: Respuesta, moduloProcesado: Modulo?) {
+        respuestas.add(respuestaProcesada)
+        modulos.add(moduloProcesado)
+    }
+}
+
+class DeteccionDemora(val demoraMinima: Int): Analizador()  {
+
+    fun respuestasDemoradasPara(unModulo: Modulo): Int {
+        var respuestasDemoradas = 0
+        respuestas.filter { r -> r.modulo == unModulo }.forEach { t ->
+            if(t.tiempo > demoraMinima){
+                respuestasDemoradas++
+            }
+        }
```

Comment:
Uy, qué difícil de leer esto. Era mucho más sencillo agregarle otra condición al filter: `&& r.tiempo > demoraMinima`.

Reply:
Pero profe, en este caso por cada respuesta demorada tenía que sumar 1 en la variable. Si pusiese esa condición en el filter, ¿donde hago la suma a la variable respuestasDemoradas por cada respuesta que cumpla la condición?

Reply:
En vez de un `filter` hacías un `count` y listo. En definitiva lo que estás haciendo es contar cuántos elementos cumplen la condición. Quedaría así:

```suggestion
        respuestas.count { r -> r.modulo == unModulo && r.tiempo > demoraMinima }
```